toolchain/check/context.h

```diff
@@ -30,6 +30,16 @@ class Context {
     SemIR::InstBlockId continue_target;
   };
 
+  // A scope in which `return` can be used.
+  struct ReturnScope {
+    // The declaration from which we can return. Inside a function, this will
+    // be a `FunctionDeclaration`.
+    SemIR::InstId decl_id;
+
+    // The value corresponding to the current `returned var`, if any.
```

Comment:
Would it be worth documenting that this can be set and unset as `returned var`s go in and out of scope?

Reply:
For me, that's clear from the "current" and "if any", but happy to add more words here if you would like?

Reply:
```suggestion
    // The value corresponding to the current `returned var`, if any. Changes during scope processing.
```

I think this is my mindset that scope information doesn't typically mutate. e.g., `decl_id` won't, right?

Reply:
Done.